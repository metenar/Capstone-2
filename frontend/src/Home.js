import React,{useContext} from "react";
import CurrentUserContext from "./CurrentUserContext"
import { Link } from "react-router-dom";
import BooksList from "./BooksList"

const Home=()=>{
    const {currentUser}=useContext(CurrentUserContext)
    return (
        <div className="text-center">
            <h1>Book API</h1>
            <p> You can follow your library using easy steps.</p>
            {currentUser 
            ? (<div>
                    <h3>Welcome Back, {currentUser.username}</h3>
                    <img className="card-img rounded-circle user-img mt-2 ml-2 p-2" style={{width:'150px'}} src={currentUser.image} alt="user avatar"/>
                    <BooksList />
                </div>
            )
        :(
            <p>
                <Link className="btn btn-primary mr-3" to="/login">
                    Login
                </Link>
                <Link className="btn btn-primary" to="/signup">
                    SignUp
                </Link>
            </p>
        )}
        </div>
    )
}
export default Home;