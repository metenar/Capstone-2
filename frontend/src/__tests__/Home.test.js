import { render,screen} from '@testing-library/react';
import { BrowserRouter, Route, Switch} from "react-router-dom";
import Home from '../Home';
import '@testing-library/jest-dom'
import CurrentUserContext from "../CurrentUserContext"

function renderHome(currentUser) {

    return render(
        <BrowserRouter>
            <CurrentUserContext.Provider value={{currentUser}}>
                <Switch>
                    <Route exect path="/">
                        <Home />
                    </Route>
                </Switch>
            </CurrentUserContext.Provider>
        </BrowserRouter>
    );
  }


test('renders without crushing', () => {
  renderHome({username:"metenar"});
});

test('should match the snapshot', () => {
  const {asFragment}=renderHome({username:"metenar"});
  expect(asFragment()).toMatchSnapshot()
});

test('should welcome back visible', ()=>{
    renderHome({username:"metenar"});
    expect(screen.getByText(/Welcome Back/)).not.toBeNull();
})