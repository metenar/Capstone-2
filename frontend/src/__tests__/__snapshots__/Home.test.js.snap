// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match the snapshot 1`] = `
<DocumentFragment>
  <div
    class="text-center"
  >
    <h1>
      Book API
    </h1>
    <p>
       You can follow your library using easy steps.
    </p>
    <div>
      <h3>
        Welcome Back, metenar
      </h3>
      <img
        alt="user avatar"
        class="card-img rounded-circle user-img mt-2 ml-2 p-2"
        style="width: 150px;"
      />
      <div
        class="col-md-8 offset-md-2"
      >
        <h3>
          Books
        </h3>
        <div
          class="mb-2"
        >
          <form
            class="form-inline"
          >
            <input
              class="form-control  flex-grow-1 form-control"
              name="term"
              placeholder="Enter search term.."
              type="text"
              value=""
            />
            <button
              class="btn btn-primary"
              type="submit"
            >
              Submit
            </button>
          </form>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
